--- pkgconf-1.9.3/libpkgconf/argvsplit.c.orig	2022-08-16 19:50:26.000000000 +0000
+++ pkgconf-1.9.3/libpkgconf/argvsplit.c	2022-11-22 02:07:03.573206600 +0000
@@ -80,7 +80,7 @@
 		if (escaped)
 		{
 			/* POSIX: only \CHAR is special inside a double quote if CHAR is {$, `, ", \, newline}. */
-			if (quote == '\"')
+			if (quote)
 			{
 				if (!(*src_iter == '$' || *src_iter == '`' || *src_iter == '"' || *src_iter == '\\'))
 					*dst_iter++ = '\\';
@@ -88,8 +88,15 @@
 				*dst_iter++ = *src_iter;
 			}
 			else
-				*dst_iter++ = *src_iter;
+			{
+				/* If we are outside a quoted string/char, an escaped space is usually used to
+				   preserve spaces in file names. */
+				if (*src_iter != ' ')
+					*dst_iter++ = '\\';
 
+				*dst_iter++ = *src_iter;
+			}
+            
 			escaped = false;
 		}
 		else if (quote)
@@ -118,11 +125,9 @@
 		}
 		else switch(*src_iter)
 		{
-#ifndef _WIN32
 			case '\\':
 				escaped = true;
 				break;
-#endif
 
 			case '\"':
 			case '\'':
